### ISSUE TITLE
`no-unused-block-params` triggers for yielded components

### ISSUE THREAD

**Issue by runspired** [ember-template-lint/ember-template-lint#2337]
example false positive

```hbs
<SortableThead as |Th|>
  <Th>col 1</Th>
  <Th>col 2</Th>
</SortableThead>
```

**Comment by lifeart** [comment]
@runspired is it full template with error? 
Tests not failing with example:

![image](https://user-images.githubusercontent.com/1360552/149285683-775ac48a-ebbc-4613-babf-d397c385c464.png)

**Comment by runspired** [comment]
There is more in the template but this is the paired down example of what triggers it. This issue triggers consistently throughout the application, not just the one template. I'll see if I can build you a reproduction.

**Comment by lifeart** [comment]
Closing as no new feedback is provided
